src/LayoutManager/types.ts

```diff
@@ -0,0 +1,148 @@
+import type {
+  BasicTransformEvent,
+  ModifiedEvent,
+  TModificationEvents,
+} from '../EventTypeDefs';
+import type { Point } from '../Point';
+import type { Group } from '../shapes/Group';
+import type { ITextEvents } from '../shapes/IText/ITextBehavior';
+import type { FabricObject } from '../shapes/Object/FabricObject';
+import type { LayoutManager } from './LayoutManager';
+import type { LayoutStrategy } from './LayoutStrategies/LayoutStrategy';
+import type {
+  LAYOUT_TYPE_INITIALIZATION,
+  LAYOUT_TYPE_ADDED,
+  LAYOUT_TYPE_IMPERATIVE,
+  LAYOUT_TYPE_REMOVED,
+  LAYOUT_TYPE_OBJECT_MODIFIED,
+  LAYOUT_TYPE_OBJECT_MODIFYING,
+} from './constants';
+
+export type LayoutTrigger =
+  | typeof LAYOUT_TYPE_INITIALIZATION
+  | typeof LAYOUT_TYPE_OBJECT_MODIFYING
+  | typeof LAYOUT_TYPE_OBJECT_MODIFIED
+  | typeof LAYOUT_TYPE_ADDED
+  | typeof LAYOUT_TYPE_REMOVED
+  | typeof LAYOUT_TYPE_IMPERATIVE;
+
+export type LayoutStrategyResult = {
+  /**
+   * new center point as measured by the **containing** plane (same as `left` with `originX` set to `center`)
+   */
+  center: Point;
+
+  /**
+   * correction vector to translate objects by, measured in the same plane as `center`
+   *
+   * Since objects are measured relative to the group's center, once the group's size changes we must apply a correction to
+   * the objects' positions so that they relate to the new center.
+   * In other words, this value makes it possible to layout objects relative to the tl corner, for instance, but not only.
+   */
+  correction?: Point;
+
+  /**
+   * correction vector to translate objects by as measured by the plane
+   */
+  relativeCorrection?: Point;
+
+  /**
+   * new width and height of the layout target
+   */
+  size: Point;
+};
+
+export type LayoutResult = {
+  result?: LayoutStrategyResult;
+  prevCenter: Point;
+  nextCenter: Point;
+  /**
+   * The vector used to offset objects by, as measured by the plane
+   */
+  offset: Point;
+};
+
+type ImperativeLayoutCommonOptions = {
+  overrides?: LayoutStrategyResult;
```

Comment:
While i look at calcBounding box can you explain me what overrides is doing?
Seems like something put there but fabricJS is not using and there is no explanation on how could be used and from who.

Reply:
It forces the layout the use these values instead of the calculated ones